src/dbus_api.rs

```diff
@@ -121,7 +121,30 @@ fn list_pools(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 
 
 fn create_volumes(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
```

Comment:
You can still follow this up with the protocol for extraction that we have arrived at:

```
if iter.arg_type() == 0 {
    return Err(MethodErr::no_args());
}
let spec: Array<(&str, &str, &str), _> = try!(iter.read::<Array<(&str, &str, &str), _>>().map_err(|_| MethodErr::invalid_arg(&0));
```

then print it out however you want.